apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/Session.cs

```diff
@@ -0,0 +1,175 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+
+namespace Google.Cloud.Spanner.V1
+{
+    public partial class Session : IDisposable
+    {
+        private readonly CancellationTokenSource _cancellation = new CancellationTokenSource();
+        private Task _keepAlive;
+        private int _lastAccessTime = Environment.TickCount;
```

Comment:
No, there'd be no need to manage DST - just always use `DateTime.UtcNow` and timespans. I really do think it's easier to reason about time-based code using time-based types, even though the .NET ones are poor. (Admittedly using `DateTime.UtcNow` there's the potential for time changing due to NTP syncs etc, but I doubt that that will be significant.)